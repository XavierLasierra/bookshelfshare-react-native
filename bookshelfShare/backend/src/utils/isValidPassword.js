const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
