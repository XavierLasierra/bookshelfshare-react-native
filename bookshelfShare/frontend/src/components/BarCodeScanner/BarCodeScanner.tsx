import React from 'react';
import { View } from 'react-native';
import { RNCamera } from 'react-native-camera';
import { SafeAreaView } from 'react-native-safe-area-context';

import CameraOverlay from '../../assets/cameraOverlay.svg';
import styles from './barCodeScanner.styles';

export default function ProductScanRNCamera() {
  interface BarCode {
    type: string,
    data: string
  }

  function onBarCodeRead(scanResult: BarCode) {
    console.warn(scanResult.type);
    console.warn(scanResult.data);
  }

  return (
    <SafeAreaView style={styles.container}>
      <RNCamera
        flashMode={RNCamera.Constants.FlashMode.auto}
        onBarCodeRead={(scanResult) => onBarCodeRead(scanResult)}
        captureAudio={false}
        androidCameraPermissionOptions={{
          title: 'Permission to use camera',
          message: 'We need your permission to use your camera',
          buttonPositive: 'Ok',
          buttonNegative: 'Cancel'
        }}
        type={RNCamera.Constants.Type.back}
        style={styles.camera}
      />
      <View style={styles.overlay}>
        <View style={styles.overlayTop} />
        <View style={styles.overlayMiddle}>
          <View style={styles.overlaySide} />
          <CameraOverlay width={300} height={250} />
          <View style={styles.overlaySide} />
        </View>
        <View style={styles.overlayBottom} />
      </View>
    </SafeAreaView>
  );
}
