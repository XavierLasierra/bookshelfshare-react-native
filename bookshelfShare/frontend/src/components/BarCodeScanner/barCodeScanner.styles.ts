import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  container: {
    flex: 1
  },
  camera: {
    height: '100%'
  },
  overlay: {
    position: 'absolute',
    top: 0,
    width: '100%'
  },
  overlayTop: {
    zIndex: 2,
    height: 80,
    backgroundColor: stylesConstants.colors.light
  },
  overlayMiddle: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  overlaySide: {
    width: 20,
    height: 350,
    backgroundColor: stylesConstants.colors.light
  },
  overlayBottom: {
    height: '100%',
    backgroundColor: stylesConstants.colors.light
  }
});
