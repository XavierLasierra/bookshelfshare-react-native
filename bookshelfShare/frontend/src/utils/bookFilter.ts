export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
