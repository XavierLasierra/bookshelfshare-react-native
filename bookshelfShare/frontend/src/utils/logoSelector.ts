import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
