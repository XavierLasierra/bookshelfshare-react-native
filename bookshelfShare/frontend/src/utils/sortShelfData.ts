export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
