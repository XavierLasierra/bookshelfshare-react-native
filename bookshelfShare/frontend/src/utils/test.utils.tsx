import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
