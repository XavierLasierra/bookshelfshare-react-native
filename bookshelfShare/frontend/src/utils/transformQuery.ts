interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
