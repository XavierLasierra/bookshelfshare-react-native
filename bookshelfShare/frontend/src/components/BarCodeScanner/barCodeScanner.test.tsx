import React from 'react';
import BarCodeScanner from './BarCodeScanner';
import { render } from '../../utils/test.utils';

const navigation = {
  push: jest.fn()
};

describe('Given an BarCodeScanner component', () => {
  describe('When it is rendered', () => {
    test('Then should match the snapshot', () => {
      const screen = render(<BarCodeScanner navigation={navigation} />);
      expect(screen).toMatchSnapshot();
    });
  });
});
