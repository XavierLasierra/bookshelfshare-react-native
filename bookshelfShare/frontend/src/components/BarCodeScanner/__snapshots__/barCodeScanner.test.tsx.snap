// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an BarCodeScanner component When it is rendered Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#D9D8C4",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      size="small"
    />
  </View>
  <View
    style={
      Object {
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#F5F5F5",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#D9D8C4",
            "borderBottomLeftRadius": 25,
            "justifyContent": "center",
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <IconMock
          height={45}
          width={45}
        />
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#D9D8C4",
              "height": 20,
              "width": 20,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F5F5F5",
                "borderTopRightRadius": 25,
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F5F5F5",
          "height": "10%",
          "zIndex": 2,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "60%",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F5F5F5",
            "height": "100%",
            "width": 20,
          }
        }
      />
      <IconMock
        height={250}
        width={300}
      />
      <View
        style={
          Object {
            "backgroundColor": "#F5F5F5",
            "height": "100%",
            "width": 20,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "height": "100%",
          "padding": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 30,
          }
        }
      >
        Scan book barcode
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "borderRadius": 10,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "fontWeight": "500",
            }
          }
        >
          Add ISBN manually
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
